### PR TITLE
Tweeks

### PR DESCRIPTION
**Pull request by HazemKhaled**
implement #28 & #29, code examples in the issues pages

### CODE REVIEW

File: controllers/widget.js [hunk 1]
```diff
@@ -153,6 +153,25 @@ function init(opts) {
...
   }
 
+	_.each(opts.children, function (child) {
+
+		// fix: https://jira.appcelerator.org/browse/TC-3583
+		if (!child) {
+			return;
+		}
+
+		console.debug('adding footer/header views from xml', child.role);
```
Comment (FokkeZB) [inline controllers/widget.js]: I guess this should be removed

File: controllers/widget.js [hunk 2]
```diff
@@ -153,6 +153,25 @@ function init(opts) {
...
+
+		switch (child.role) {
+		  case 'headerView':
+		    $.table.headerView = child;
+		    break;
+		  case 'footerView':
+		    $.table.footerView = child;
+		    break;
+		}
+	});
```
Comment (FokkeZB) [inline controllers/widget.js]: You probably want to unset `opts.children` once processed so you don't have references to them anymore.

File: controllers/widget.js (hunk 1 above)
Comment (HazemKhaled) [inline controllers/widget.js]: My case
```xml
<Widget id="form" src="nl.fokkezb.form" config="forms/project_add">
    <View class="headerView" role="headerView" formFactor="tablet">
        <Label id="titleLabel"/>
    </View>
    <Button/>
    <View class="footerView" role="footerView">
        <Button id="sendButton" onClick="sendButtonClicked"/>
    </View>
</Widget>
```

child of zero come with null, maybe TC-3583 fixed on platform but not with formfactor

Comment (HazemKhaled) [inline controllers/widget.js]: issue updated

File: controllers/widget.js (hunk 2 above)
Comment (HazemKhaled) [inline controllers/widget.js]: Yes that's better, push now

**Pull request merged**